### PR TITLE
Categories to groups

### PR DESCRIPTION
**Pull request by IanPhilips**
- Categories are now default groups that people are added to when they sign up
- These new default groups created by Manifold Markets don't have chat walls: those are for user-created groups only
- Current users are added to these default groups unless they unchecked any categories when those were available previously, in this case they are added only to the cateogires they were following
- Given we don't have free markets anymore I hypothesize we don't need to filter out personal and fun markets when a user signs up/or is signed out. 
  - Addendum: signed out users should see markets related to the market they landed on via similar bettors' betting history
- The personal filter filters for your followers' markets and markets in the groups you're in
Screen Shot 2022-07-12 at 11 42 50 AM
Screen Shot 2022-07-12 at 11 45 09 AM

### CODE REVIEW

**Comment by IanPhilips** [comment]
By @mantikoros 's request I added a uniqueBettors attribute to `contract.ts` and taking @mqp 's advice, I now calculate this in the trigger `on-create-bet.ts`. Now new users will see the most popular markets by unique bettors. This isn't backfilled however so it would bias towards currently active, popular markets
Screen Shot 2022-07-12 at 4 40 41 PM

**Comment by IanPhilips** [comment]
Perhaps I could rename the index to most popular?

**Comment by jahooma** [comment]
+1 for Most popular

**Comment by jahooma** [comment]
One problem with unique bettors is that it becomes static very quickly. Something like unique bettors in the last week might be better, as a suggestion.

**Comment by IanPhilips** [comment]
Oh interesting idea! Hm, should we leave it and see how stale it gets or just change it now?

**Comment by jahooma** [comment]
Oh, we could leave it for now I guess. Might have a different idea in a week or two

File: functions/src/on-create-bet.ts [hunk 1]
```diff
@@ -26,8 +39,99 @@ export const onCreateBet = functions.firestore
...
+  if (!contract) {
+    log(`Could not find contract ${contractId}`)
+    return
+  }
+  const contractBets = (
+    await firestore
+      .collection(`contracts/${contractId}/bets`)
+      .where('userId', '!=', contract.creatorId)
+      .get()
+  ).docs.map((doc) => doc.data() as Bet)
```
Comment (jahooma) [inline functions/src/on-create-bet.ts:63]: Seems like this won't scale well. Fetching all bets (except for the creator) on every bet is n^2.

Can't you just check if the new bet is a new unique bettor compared to the saved uniqueBettors field? If the uniqueBettors field doesn't exist then you could fetch all the bets.

**Comment by IanPhilips** [comment]
Oh schniz that would actually be great so I could add an additional sort
for the number of followed user ids in the unique bettors array!

**Comment by IanPhilips** [comment]
Okay so now the personal filter will show:
- contracts in groups that the user is a member of
- contracts created by users that the user follows
- contracts bet on by users that the user follows
- contracts bet on by the user

**Comment by IanPhilips** [comment]
What should the personal filter be named?
- Recommended
- Interesting
- Personalized
- Something else?

**Comment by jahooma** [comment]
Not sure. Maybe For you?

**Pull request merged**